vmas/scenarios/transport.py

```diff
@@ -166,20 +170,28 @@ def reward(self, agent: Agent):
                     Color.GREEN.value, device=self.world.device, dtype=torch.float32
                 )
 
-                package_shaping = package.dist_to_goal * self.package_goal_dist_reward_factor
-                self.rew[~package.on_goal] += (
-                    package.global_shaping[~package.on_goal]
-                    - package_shaping[~package.on_goal]
-                )
-                package.global_shaping = package_shaping
+                # dense reward
+                if not self.sparse_reward_only:
+                    package_shaping = package.dist_to_goal * self.package_goal_dist_reward_factor
+                    self.rew[~package.on_goal] += (
+                        package.global_shaping[~package.on_goal]
+                        - package_shaping[~package.on_goal]
+                        )
+                    package.global_shaping = package_shaping
+                
+                # sparse reward. Only get rewarded for getting the package on goal.
+                else:
+                    self.rew[package.on_goal] += 1.0
+                
 
         # reward for how close agents are to all packages
-        for i, package in enumerate(self.packages):
-            dist_to_pkg = torch.linalg.vector_norm(agent.state.pos - package.state.pos, dim=-1)
-            # any small distance gets "floored"
-            # dist_to_pkg[dist_to_pkg < 0.1] = 0.1
+        if not self.sparse_reward_only:
```

Comment:
nit: I think it'd be cleaner if you simply flip the boolean to `self.add_dense_rewards`, took me a second to understand what "not sparse reward only" meant

Reply:
Makes sense, I will add that change! I agree.